### PR TITLE
Adding the AWS CLI v2.

### PR DESCRIPTION
**Pull request by lazzurs**
This is a cask for the AWS CLI v2 as this is now packaged as a pkg
rather than a python module to be installed with pip.

Tested in a tap and appears to work as it should.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/awscli.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cask 'awscli' do
+  version '2.0.0'
+  sha256 'a99c26cd2b5655970db91259a247f585aea32cfb7347e1a28a25b6f6484b16dc'
+
+  # awscli.amazonaws.com was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/awscli.rb]: This comment needs to be standard.

**Comment by vitorgalvao** [comment]
If this already exists as a formula, what’s the point of the cask?

**Comment by lazzurs** [comment]
Moving this to a cask as the move from version 1 to version 2 moves the awscli from a python package that can be installed with pip to a macOS package. They have done this because they are now bundling python and everything else the cli needs with the cli tool. This is a packaging change by AWS.

**Comment by vitorgalvao** [comment]
> Moving this to a cask

Then the `conflicts_with` line is unnecessary, as the formula will have to be removed. Have you submitted the PR to do so?

File: Casks/awscli.rb (hunk 1 above)
Comment (lazzurs) [reply Casks/awscli.rb]: Fixed.

**Comment by lazzurs** [comment]
Added a bug and pull request to remove the formula from core. Will remove the conflicts_with now.

**Comment by lazzurs** [comment]
The builds are failing because of the conflict with the existing formula. This should be complete and thanks for all of the help on it.

If there is anything more I can do please let me know.

**Comment by vitorgalvao** [comment]
Everything looks good, thank you. Now we just have to wait on the formula PR.

**Comment by fxcoudert** [comment]
From the comments at https://github.com/Homebrew/homebrew-core/pull/50056, it appears the CLI is open source and should therefore be installed from source as a formula, rather than in cask.

**Comment by vitorgalvao** [comment]
Perfect, I was expecting that to be the case. Closing this PR for now. If anything changes, please do let me know and we’ll revisit.

**Pull request closed without merging**